packages/docusaurus-plugin-content-blog/src/index.ts

```diff
@@ -512,14 +513,11 @@ export default async function pluginContentBlog(
       };
     },
 
-    async postBuild({outDir}: Props) {
+    async postBuild(this: LoadedPlugin<BlogContent>, {outDir}: Props) {
       if (!options.feedOptions.type) {
         return;
       }
-
-      // TODO: we shouldn't need to re-read the posts here!
-      // postBuild should receive loadedContent
-      const blogPosts = await generateBlogPosts(contentPaths, context, options);
+      const {blogPosts} = this.content;
```

Comment:
Let's not create a new temp variable for the sole purpose of injecting `content` and actually acknowledge the existence of `this`. We will document this in the near future because `this` is quite intuitive in this context.

Reply:
I'd rather inject the content inside `postBuild`: this is something I want to do for a while now

Reply:
Fair, since we also inject `content` in `configureWebpack`. But FWIW, I like the Webpack loader model more, where everything's accessible through `this`. This means less temp variables and a clearer mental model (plugins are stateful, not stateless)

Reply:
> plugins are stateful, not stateless

🤷‍♂️ React components too, and yet we use hooks instead of this :D State can be provided more explicitly.

And the state of a plugin depends on its lifecycle. 

For example `this.content `is not accessible before loadContent() has been called: you can't easily prevent a mistake with TS when using `this`.